packages/template-block/src/TemplateBlock.tsx

```diff
@@ -61,98 +53,77 @@ export const TemplateBlock = ({ appBridge }: BlockProps): ReactElement => {
 
+    if (!selectedTemplate && !isEditing) {
+        return <div data-test-id="container" className="template-block"></div>;
+    }
+
     return (
-        <div data-test-id="template-block-container" className="template-block">
-            {selectedTemplate || isEditing ? (
-                <div
-                    data-test-id="template-block-card"
-                    className="tw-border tw-border-black-20"
-                    style={{
-                        ...(hasBackground && getBackgroundColorStyles(backgroundColor)),
-                        borderRadius,
-                        border,
-                        padding: getCardPadding(blockSettings),
-                    }}
-                >
-                    {isEditing && lastErrorMessage !== '' && <AlertError errorMessage={lastErrorMessage} />}
-                    <div
-                        data-test-id="template-block-content"
-                        className={`tw-flex ${flexDirectionStyles}`}
-                        style={{
-                            gap: GAP,
-                            alignItems: isRows ? textAnchoringVertical : undefined,
-                        }}
-                    >
-                        {hasPreview && (
-                            <TemplatePreview
+        <div data-test-id="container" className="template-block">
+            <div data-test-id="card" style={cardStyles}>
+                {isEditing && lastErrorMessage !== '' && <AlertError errorMessage={lastErrorMessage} />}
+
+                <div data-test-id="content" className={contentClasses}>
+                    {hasPreview && (
+                        <TemplatePreview
+                            appBridge={appBridge}
+                            blockSettings={blockSettings}
+                            template={selectedTemplate}
+                            previewClasses={previewClasses}
+                            updateBlockSettings={updateBlockSettings}
+                            onOpenTemplateChooser={handleOpenTemplateChooser}
+                        />
+                    )}
+
+                    <div data-test-id="text-cta-wrapper" className={textCtaWrapperClasses}>
+                        <div data-test-id="text" className={textClasses}>
+                            <TemplateText
+                                appBridge={appBridge}
+                                updateBlockSettings={updateBlockSettings}
+                                title={title}
+                                description={description}
+                                pageCount={
+                                    blockSettings.hasPageCount !== false
+                                        ? selectedTemplate?.pages.length ?? 0
+                                        : undefined
+                                }
+                                isEditing={isEditing}
+                                key={templateTextKey}
```

Comment:
```suggestion
                                templateTextKey={templateTextKey}
```
![Screenshot 2024-01-24 at 08 55 15](https://github.com/Frontify/guideline-blocks/assets/9362990/d1da8e11-04a1-4874-9d95-f95769289b08)

Reply:
[Reserved keyword](https://legacy.reactjs.org/warnings/special-props.html), thanks for the hint. 🧑‍🎓